<?xml version="5.0"?>
<launch>

<arg name="rate" default="1.0" />

<arg name="meshfile" default="/home/amock/datasets/oru4.dae" />
<!-- <arg name="bagfile" default="/home/amock/datasets/micp_mulran/KAIST/mulran_kaist01_micp_lvr2.bag" /> -->
<!-- <arg name="bagfile" default="/home/amock/datasets/micp_mulran/KAIST/mulran_kaist02_micp_lvr2.bag" /> -->
<!-- <arg name="bagfile" default="/home/amock/datasets/micp_mulran/KAIST/mulran_kaist03_micp_lvr2.bag" /> -->

<!-- <arg name="meshfile" default="/home/amock/datasets/mulran_meshes/lvr2/DCC/dcc03_mesh_red05_cleaned.ply" /> -->
<!-- <arg name="bagfile" default="/home/amock/datasets/micp_mulran/DCC/mulran_dcc01_micp_lvr2.bag" /> -->
<!-- <arg name="bagfile" default="/home/amock/datasets/micp_mulran/DCC/mulran_dcc02_micp_lvr2.bag" /> -->
<!-- <arg name="bagfile" default="/home/amock/datasets/micp_mulran/DCC/mulran_dcc03_micp_lvr2.bag" /> -->


<arg name="radarparams" default="$(find radarays_ros)/cfg/mulran_kaist_dyncfg.yaml" />
<!-- <arg name="radarparams" default="$(find radarays_ros)/cfg/mulran_kaist_dyncfg_laserlike.yaml" /> -->


<node pkg="radarays_ros" type="radar_simulator" name="radar_simulator" output="screen">
    <param name="map_file" type="string" value="$(arg meshfile)" />
    <param name="map_frame" type="string" value="map" />
    <!-- <param name="sync_topic" type="string" value="/Navtech/Polar" /> -->
    <param name="sensor_frame" type="string" value="navtech" />
    # Load inital materials
    <rosparam command="load" file="$(find radarays_ros)/config/oru4.yaml" />
    <param name="node_mode" type="string" value="server" />
</node>

<node name="dynamic_reconfigure_load" pkg="dynamic_reconfigure" type="dynparam" 
    args="load /radar_simulator $(arg radarparams)" />

</launch>
